devito/data/decomposition.py

```diff
@@ -313,16 +313,20 @@ def index_glb_to_loc(self, *args, rel=True):
         elif len(args) == 2:
             # index_glb_to_loc(offset, side)
             if self.loc_empty:
-                return 0
-            rel_ofs, side = args
+                return None
+            abs_ofs, side = args
             if side is LEFT:
-                abs_ofs = self.glb_min + rel_ofs
-                size = self.loc_abs_max - base + 1
-                return min(abs_ofs - base, size) if abs_ofs > base else 0
+                rel_ofs = self.glb_min + abs_ofs - base
+                if abs_ofs >= base and abs_ofs <= top:
+                    return rel_ofs
+                else:
+                    return None
             else:
-                abs_ofs = self.glb_max - rel_ofs
-                size = top - self.loc_abs_min + 1
-                return min(top - abs_ofs, size) if abs_ofs < top else 0
+                rel_ofs = abs_ofs - (self.glb_max - top)
+                if abs_ofs >= self.glb_max - top and abs_ofs <= self.glb_max - base:
+                    return rel_ofs
+                else:
+                    return None
```

Comment:
Looks fine to me. Small question, does this incur some performance slow down? This is called a lot in sparse with MPI (for each coordinate individually). Doesn't look like it would but  just checking

Reply:
The MPI build is currently taking ages, will investigate to see if this could be the reason.

Reply:
MPI taking ages for quite a while now, suddently doubled time at some point. Still not sure why, so check with previous build if slower or just normal slow.

Reply:
Checked against some previous build and they were ~80 mins. This is at 100 and still going...